### ISSUE TITLE
The predicted probabilities of some issues are 99.99%

### ISSUE THREAD

**Issue by mcxwx123** [osslab-pku/gfi-bot#44]
Currently, many open issues in some projects have a GFI probability of 99.99%, and some of these issues clearly should not be marked as GFI.
![image](https://user-images.githubusercontent.com/54138388/197691019-154b862c-d001-4345-a3ee-a636a85946ee.png)
The performance metric of the model is also unusually high.
![image](https://user-images.githubusercontent.com/54138388/197690958-e03661c7-4ec7-4b28-a1c4-192cf3e19eee.png)
I examined the code and found two features that may be problematic. The first is 'created_at_timestamp', which is not one of the features and should not be included in X (def get_x_y() in gfibot/model/utils.py). The second one is 'rpt_gfi_ratio', when I try to drop this feature, the model performance metrics appear to drop significantly. 

The problems can be solved by the following steps:

1. Add 'created_at_timestamp' to https://github.com/osslab-pku/gfi-bot/blob/7ed07616507578e78856cc45c099602648b81932/gfibot/model/utils.py#L135
2. The gfi_ratio and gfi_num features should be calculated with a new issue list, which only includes issues closed before the data collection time. https://github.com/osslab-pku/gfi-bot/blob/7ed07616507578e78856cc45c099602648b81932/gfibot/model/dataloader.py#L112 https://github.com/osslab-pku/gfi-bot/blob/7ed07616507578e78856cc45c099602648b81932/gfibot/model/dataloader.py#L118 
Now we use the following list. A new issues list like `issues = [i for i in user.issues if i.closed_at <= t]` should be created for calculating gfi_ratio and gfi_num later.
https://github.com/osslab-pku/gfi-bot/blob/7ed07616507578e78856cc45c099602648b81932/gfibot/data/dataset.py#L205

There may be a situation where most of the prediction probabilities are close to 0 after the above features are corrected because of the imbalance of positive and negative instances in the training data, which can be solved by balancing the training dataset using methods such as SMOTE and ADASYN. Then we can check whether the '99.99% probabilities' problem is solved.

**Comment by arsyadal** [comment]
Hi @mcxwx123 👋, I have resolved the 99.99% probability issue by dropping \created_at_timestamp\ and \closed_at_timestamp\ from the feature matrix \X\ in \get_x_y()\ (\gfibot/model/utils.py\), and adding explicit non-null \closed_at\ checks when filtering resolved issues before timestamp \	\ in \gfibot/data/dataset.py\. The fix has been submitted in PR #69.
